statics/js/components/rule-detail.js

```diff
@@ -0,0 +1,494 @@
+ * @return the port or ANY_PORT if not found.
+*/
+function inport(filters) {
+  var matchInport = reFindInport.exec(filters);
+  return matchInport ? safePort(matchInport[2]) : ANY_PORT;
+}
+
+/** Summarize the filter of a rule, filling the inPort
+ *  @param rule: the rule to complete
+ */
+function summarizeFilter(rule) {
+  rule.inPort = inport(rule.filters);
+}
+
+/** Computes the summary of a rule, both filters and outActions
+ * @param rule: the rule to complete
+ */
+function summarize(rule) {
+  summarizeFilter(rule);
+  summarizeActions(rule);
+}
+
+/** Classify the eleements of an array into a table according to a classifier function.
+ *
+ * A generic function that taken a classifier that gives back
+ * the kind of an object as an integers transform an array of objects
+ * in a table indexed by integers of list where each list is the set of elements of the
+ * array that have the key as classifier.
+ * @param array: the array to classify
+ * @param classifier: the classifier function
+ * @return the result as an integer indexed table of list
+ */
+function classify(array, classifier) {
+  var result = {};
+  for (var i = 0; i < array.length; i++) {
+    var elem = array[i];
+    var key = classifier(elem);
+    var list = result[key];
+    if (list !== undefined)
+      list.push(elem);
+    else
+      result[key] = [elem];
+  }
+  return result;
+}
+
+/** Computes the chain of node that represent the path of node to highlight for a given bridge interface
+ *
+ * If it is an external port, we only go up to the port itself, if it is a patch port, we continue on the other bridge
+ * but the bridge node is not part of the highlighted path. itfs is filled with an entry for the port using the numbering of ovs-ofctl as key
+ * and the list of nodes as values.
+ * @param c a GNode of type ovsport
+ * @param graph the full graph model
+ * @param itfs the table to fill with information on the port represented.
+*/
+function extractPort(c, graph, itfs) {
+  var pot = graph.getTargets(c);
+  for (var i = 0; i < pot.length; i++) {
+    var cc = pot[i];
+    var ofport = cc.metadata.OfPort;
+    if (ofport === undefined)
+      continue;
+    var itf = [c, cc];
+    itfs[ofport] = itf;
+    if (cc.metadata.Type === 'patch') {
+      var ccc = graph.getNeighbor(cc, 'patch');
+      if (ccc === undefined)
+        return;
+      itf.push(ccc);
+      var cccc = graph.getNeighbor(ccc, 'ovsport');
+      if (cccc !== undefined)
+        itf.push(cccc);
+    }
+  }
+}
+
+/** A representation of the rules associated to a bridge and the associated functions.
+ *  The methods of the rule-detail and rule-table-detail components are in fact implemented
+ *  by this object
+*/
+var BridgeLayout = (function () {
+
+  /** Builds the bridge layout
+   * @param graph the global graph hosting the bridge
+   * @param bridge the bridge node whose rules are represented.
+   */
+  function BridgeLayout(graph, bridge, store) {
+    this.graph = graph;
```

Comment:
Something that could be really nice would be to listen for graph event so that we could refresh the rule without to unselect/reselect the node. Nothing mandatory for now but that something we will have to do.

Reply:
Done. With also a way to unregister the callback at the end of the component lifecycle.